src/plugins/intel_gpu/tests/test_cases/scatter_nd_update_gpu_test.cpp

```diff
@@ -12,17 +14,1154 @@
 #include <intel_gpu/graph/network.hpp>
 
 #include <cstddef>
+#include <cstring>
+#include <numeric>
+#include <stdlib.h>
+#include <time.h>
+#include <algorithm>
 
 using namespace cldnn;
 using namespace ::tests;
 
 
+struct scatter_nd_update_basic_test_params
+{
+    data_types input_type;
+    data_types indices_type;
+    data_types updates_type;
+    format input_format;
+    format indices_format;
+    format updates_format;
+    format input_result_format;
+    format indices_result_format;
+    format updates_result_format;
+    tensor input_size;
+    tensor indices_size;
+    tensor updates_size;
+    int indices_rank;
+};
+
+struct scatter_nd_update_random_test : testing::TestWithParam<scatter_nd_update_basic_test_params>
+{
+    template<typename T>
+    std::vector<T> generate_random_indices(size_t a, int min, int max, int k = 1) 
+    {
+        srand(time(0));
+        std::vector<T> vec(a);
+        for(int i = 0; i < a; ++i) {
+            int val = rand() % (max + 1);
+            if (val >= min && std::find(vec.begin(), vec.end(), static_cast<T>(val)) == vec.end())
+                vec[i] = val;
+        }
+        return vec;
+    }
+
+    template<typename T, typename T_size>
+    void execute_fp16(const scatter_nd_update_basic_test_params& params)
```

Comment:
Is this needed ? cant we merge it with execute?

Reply:
Merged and updated :)